Sources/Sentry/SentryUIViewControllerPerformanceTracker.m

```diff
@@ -469,11 +466,104 @@ - (void)measurePerformance:(NSString *)description
     }
 }
 
-- (void)reportInitialDisplay:(UIViewController *)controller
+- (void)reportInitialDisplayForController:(NSObject *)controller
+{
+    SENTRY_LOG_DEBUG(
+        @"[UIViewController Performance] Reporting initial display for controller: %@", controller);
+    if (self.currentTTDTracker == nil) {
+        SENTRY_LOG_DEBUG(@"[UIViewController Performance] Can't report initial display, no screen "
+                         @"transaction being tracked right now.");
+        return;
+    }
+    [self.currentTTDTracker reportInitialDisplay];
+    SENTRY_LOG_DEBUG(
+        @"[UIViewController Performance] Reported initial display for controller: %@", controller);
+}
+
+// - MARK: - Getter and Setter Helpers
+
+- (SentryTimeToDisplayTracker *_Nullable)getTimeToDisplayTrackerForController:
+    (UIViewController *)controller
+{
+    SENTRY_LOG_DEBUG(
+        @"[UIViewController Performance] Getting time to display tracker for controller: %@",
+        controller);
+    // Use a global map to store the time to display tracker to avoid memory issues with associated
+    // objects.
```

Comment:
`l`: I think mentioning once that we had memory issues with associated object at the top of the class is enough. IMO, we can remove these comments here. It's mostly noise.